semver.py

```diff
@@ -493,8 +519,10 @@ def format_version(major, minor, patch, prerelease=None, build=None):
     >>> import semver
     >>> semver.format_version(3, 4, 5, 'pre.2', 'build.4')
     '3.4.5-pre.2+build.4'
+    >>> semver.format_version("3", 4, "5", 'pre.2', 'build.4')
+    '3.4.5-pre.2+build.4'
     """
-    version = "%d.%d.%d" % (major, minor, patch)
+    version = "%d.%d.%d" % (int(major), int(minor), int(patch))
```

Comment:
maybe just `'{}.{}.{}'.format(major, minor, patch)`? No need to convert str->int->str
Alternative: Create a Version object (constructor handles conversion now) and do the formatting based on that object.

Reply:
> maybe just '{}.{}.{}'.format(major, minor, patch)? No need to convert str->int->str

That's a good idea! :+1: 

> Alternative: Create a Version object (constructor handles conversion now) and do the formatting based on that object.

That would change the implementation of `format_version` which IMHO is something for another PR.

Reply:
Well you added the support for string arguments in this PR. Then maybe extract that to another PR too and implement it "properly"? This avoids changing this back-and-forth

Reply:
Well, the `semver.replace(version, **parts)` delegates the formatting to the `format_version` function. If we want to allow not only integers, but also strings in the parts argument,  `format_version` needs to accept also strings.

It's really this tiny change to make str -> int in the above line 525. It wasn't my intention to change the whole implementation of that function.

If I move that change into another PR, this PR depends on the other. Which would complicate things a bit, do you agree? :wink:

Reply:
If `semver.replace` uses `VersionInfo.replace` then you don't need that change as the ctor will make an int out of the str in `parts`. Problem solved ;)